tests/unit/acquisition/function/test_function.py

```diff
@@ -965,6 +966,144 @@ def prepare_acquisition_function(
     npt.assert_allclose(eci(x), ei(x) * pof(x))
 
 
+@pytest.mark.parametrize("sample_size", [-2, 0])
+def test_batch_expected_improvement_raises_for_invalid_sample_size(
+    sample_size: int,
+) -> None:
+    with pytest.raises(tf.errors.InvalidArgumentError):
+        BatchExpectedImprovement(sample_size=sample_size)
+
+
+@pytest.mark.parametrize("sample_size", [2])
+@pytest.mark.parametrize("jitter", [-1e0])
+def test_batch_expected_improvement_raises_for_invalid_jitter(
+    sample_size: int,
+    jitter: float,
+) -> None:
+    with pytest.raises(tf.errors.InvalidArgumentError):
+        BatchExpectedImprovement(sample_size=sample_size, jitter=jitter)
+
+
+@pytest.mark.parametrize("sample_size", [100])
+@pytest.mark.parametrize("jitter", [1e-6])
+def test_batch_expected_improvement_raises_for_empty_data(
+    sample_size: int,
+    jitter: float,
+) -> None:
+    builder = BatchExpectedImprovement(
+        sample_size=sample_size,
+        jitter=jitter,
+    )
+    data = Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))
+    matern52 = tfp.math.psd_kernels.MaternFiveHalves(
+        amplitude=tf.cast(2.3, tf.float64), length_scale=tf.cast(0.5, tf.float64)
+    )
+    model = GaussianProcessWithBatchSamplers(
+        [lambda x: Branin.objective(x), lambda x: quadratic(x)], [matern52, rbf()]
+    )
+    with pytest.raises(tf.errors.InvalidArgumentError):
+        builder.prepare_acquisition_function(model, dataset=data)
+    with pytest.raises(tf.errors.InvalidArgumentError):
+        builder.prepare_acquisition_function(model)
+
+
+@pytest.mark.parametrize("num_data", [5])
```

Comment:
Should we try a small number of different values to make sure this test is strong enough? also, sample size is suspiciously small, i would expect large error from MC-EI, so maybe we're not testing a sensitive case here?

Reply:
also don't know if we care about having a large number of `@pytest.mark.parametrize` that are useless since we're testing a single value. @uri-granta does it matter?

Reply:
Yes, I'll add a few more values to make sure the test is strong enough.

The sample size for MC-EI is `10000`, which seems already quite large. The sample size for the Genz EI is `100`, which is much smaller but this is the point about the Genz method, i.e. that it can get very accurate approximations with far fewer samples.

Reply:
oh sorry read it too quickly, didn't see the mc_sample_size=1e5